src/main/java/com/amazon/opendistroforelasticsearch/sql/query/AggregationQueryAction.java

```diff
@@ -231,10 +238,34 @@ private AggregationBuilder getGroupAgg(Field field, Select select2) throws SqlPa
         }
 
         if (!refrence) lastAgg = aggMaker.makeGroupAgg(field);
-        
+
+        // find if we have order for that aggregation. As of now only special case for script fields
+        if (field.isScriptField()) {
+            addOrderByScriptFieldIfPresent((ScriptMethodField) field, select, lastAgg);
+        }
+
         return lastAgg;
     }
 
+    private void addOrderByScriptFieldIfPresent(ScriptMethodField field, Select select, AggregationBuilder lastAgg) {
```

Comment:
I think it's more consistent with your method name to "narrow down" the second param and pass in `select.getOrderBys()`.

Reply:
Done